File: src/Other/Input.py
import Blackprint
import types

@Blackprint.registerNode('MyCustom/Other/Input')
class Simple(Blackprint.Node):
	output = {
		'Value': float,
	}

	def __init__(this, instance):
		Blackprint.Node.__init__(this, instance)

		iface = this.setInterface('BPIC/YourModuleName/Other/Input')
		iface.title = "Input"

	# Bring value from imported iface to node output
	def imported(this, data):
		if data == None: return

		val = data['value']
		this.iface.data.value = val
		this.output.Value = val

	# Remote sync in
	def syncIn(this, id, data):
		if(id == 'data'):
			this.iface.data.value = data.value
			this.iface.changed(data.value)

class InputIFaceData:
	def __init__(this, iface):
		this._iface = iface
		this._data = {"value": '...'}

	@property
	def value(this):
		return this._data['value']

	@value.setter
	def value(this, val):
		this._data['value'] = val
		this._iface.changed(val)

@Blackprint.registerInterface('BPIC/YourModuleName/Other/Input')
class InputIFace(Blackprint.Interface):
	def __init__(this, node):
		super(InputIFace, this).__init__(node)
		this.data = InputIFaceData(this)

	def changed(this, val):
		node = this.node

		# This node still being imported
		if(this.importing != False):
			return

		node.output['Value'] = float(val)
		node.syncOut('data', {'value': this.data.value})

File: src/Other/Input.js
/**
 * Just a simple input box
 * @blackprint node
 */
Blackprint.registerNode('YourModuleName/Other/Input',
class extends Blackprint.Node {
	static output = {
		Value: Number,
	};

	constructor(instance){
		super(instance);

		let iface = this.setInterface('BPIC/YourModuleName/Other/Input');
		iface.title = "Input";
	}

	// Bring value from imported node to handle output
	imported(data){
		let iface = this.iface;
		if(data === undefined) return;

		Object.assign(iface.data, data);
		this.output.Value = data.value;
	}
});

Blackprint.registerInterface('BPIC/YourModuleName/Other/Input',
Context.IFace.Input = class InputIFace extends Blackprint.Interface{
	constructor(node){
		super(node);
		this.data = new ExampleInputData(this);
	}

	changed(text, ev){
		let node = this.node;

		// This node still being imported
		if(this.importing !== false)
			return;

		// iface.data.value === text;
		node.output.Value = this.data.value;
	}
});

class ExampleInputData {
	// Use underscore "_" or "$" to avoid being exported as JSON
	#value = 0;
	#iface = null;

	constructor(iface){
		this.#iface = iface;
	}

	get value(){ return this.#value }
	set value(val){
		if(this.#value === val) return;
		this.#value = val;
		this.#iface.changed(val);
		this.#iface.node.routes.routeOut();
	}
}

Blackprint.utils.setEnumerablePrototype(ExampleInputData, {
	value: true,
});